src/components/Signup.tsx

```diff
@@ -1,28 +1,50 @@
-import React, { useState, ChangeEvent, FormEvent } from "react";
+import { useState, FormEvent } from "react";
 import { signupFields } from "../constants/formFields";
 import FormAction from "./FormAction";
 import Input from "./Input";
-
+import apiInstance from "../axios";
+import SuccessSnackbar from "./SuccessResponseSnackbar";
+import FailureSnackbar from "./FailureResponseSnackbar";
 const fields = signupFields;
 let fieldsState: { [key: string]: string } = {};
 fields.forEach((field) => (fieldsState[field.id] = ""));
 
 const Signup = () => {
   const [signupState, setSignupState] = useState(fieldsState);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (e: { target: { id: any; value: any } }) => {
     setSignupState({ ...signupState, [e.target.id]: e.target.value });
   };
-
+  const [response, setResponse] = useState("");
+  const [showSnackbar, setShowSnackbar] = useState(false);
+  const [isSuccess, setIsSuccess] = useState(false);
   const handleSubmit = (e: FormEvent) => {
     e.preventDefault();
     console.log(signupState);
     createAccount();
   };
-
-  // Handle Signup API Integration here
-  const createAccount = () => {
-    // API integration logic
+  console.log(signupState);
+  const signUpReqData = {
+    name: signupState.username,
+    email: signupState.email,
+    password: signupState.password,
+    address: signupState.address,
+  };
```

Comment:
This might be working fine due to the way react hooks work but it does not seem right. Improve this by making it a function like
```suggestion
  const getSignUpReqData = () =>  ({
    name: signupState.username,
    email: signupState.email,
    password: signupState.password,
    address: signupState.address,
  });
```

Reply:
done